src/Raven.Server/Documents/Sharding/ShardLocator.cs

```diff
@@ -1,10 +1,26 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
 using Raven.Server.ServerWide.Context;
 
 namespace Raven.Server.Documents.Sharding
 {
     public static class ShardLocator
     {
+        public static DocumentIdsByShardIndex GroupIdsByShardIndex(IEnumerable<string> ids,
+            ShardedContext shardedContext, TransactionOperationContext context)
+        {
+            var result = new DocumentIdsByShardIndex();
+
+            foreach (var id in ids)
+            {
+                var shardIndex = shardedContext.GetShardIndex(context, id);
+                result.Add(shardIndex, id);
+            }
+
+            return result;
+        }
+
         public static Dictionary<int, List<int>> GetDocumentIdsShards(IList<string> ids,
```

Comment:
do we need this now?

Reply:
this is different from the `GroupIdsByShardIndex` which accepts `IEnumerable<Slice>` now